### PR TITLE
Added 'object' and 'assembly' override shaders

### PR DESCRIPTION
**Pull request by bassemhermina**
Solving issue #2420 
"Objects" override shader can be easily tested by instantiating any of the objects in the cornelBox project as it's instantiated so far and large-scaled so colors can be easily compared. (is that an issue?)
but for "Assemblies" shader, instantiate the assembly with name "assembly_inst1" then run this to translate it a little bit for it to be visible.
(running the python code while interactive render is activated crashes appleseed.studio instantly on linux ubuntu 16.04, is that an issue too?)

from appleseed.studio import *
import appleseed as asr
scene = current_project().get_scene()

assembly = scene.assembly_instances().get_by_name("assembly_inst1")
matrix = asr.Matrix4d([1.0, 0.0, 0.0, 0.3,
                       0.0, 1.0, 0.0, 0.3,
                       0.0, 0.0, 1.0, 0.0,
                       0.0, 0.0, 0.0, 1.0])
assembly.transform_sequence().set_transform(0.0, asr.Transformd(matrix))

### CODE REVIEW

File: src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp [hunk 1]
```diff
@@ -576,11 +580,23 @@ void DiagnosticSurfaceShader::evaluate(
         }
         break;
 
+      case Assemblies:
+        set_shading_result(
+            shading_result,
+            integer_to_color3<float>(shading_point.get_assembly_instance().find_assembly()->get_uid()));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp]: Use `ShadingPoint::get_assembly()` which is very fast. `AssemblyInstance::find_assembly()` is very slow, it's used to find the assembly outside of rendering.

File: src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp [hunk 2]
```diff
@@ -576,11 +580,23 @@ void DiagnosticSurfaceShader::evaluate(
...
       case AssemblyInstances:
         set_shading_result(
             shading_result,
             integer_to_color3<float>(shading_point.get_assembly_instance().get_uid()));
         break;
+      
+      case Objects:
+        set_shading_result(
+            shading_result,
+            integer_to_color3<float>(shading_point.get_object_instance().find_object()->get_uid()));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp]: Similarly, use `ShadingPoint::get_object()`.

File: src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp [hunk 3]
```diff
@@ -583,7 +583,7 @@ void DiagnosticSurfaceShader::evaluate(
       case Assemblies:
         set_shading_result(
             shading_result,
-            integer_to_color3<float>(shading_point.get_assembly_instance().find_assembly()->get_uid()));
+            integer_to_color3<float>(shading_point.get_assembly_instance().get_assembly().get_uid()));
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp]: Not what I suggested, please read again my comments :)

Comment (bassemhermina) [reply src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp]: ahh ahh ah sorry for that! fixing it now!

Comment (bassemhermina) [reply src/appleseed/renderer/modeling/surfaceshader/diagnosticsurfaceshader.cpp]: done! sorry still getting used to the codebase :)

**Pull request merged**